sympy/functions/elementary/hyperbolic.py

```diff
@@ -336,12 +346,25 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import pi
```

Comment:
Imports should be at the top unless there are circular dependency issues, in which case you can move the import into a function.